### PR TITLE
Feature map component

### PR DESCRIPTION
**Pull request by mitchthorson**
### What's in this pull request

- [x] New component/feature

### Description

This PR adds a new `SVGMap` component and several related sub-components, `PolygonLayer`, `PointLayer` and `LabelLayer` and associated documentation that explains how to use these with several different examples.

### Before submitting, please check that you've

- [x] Formatted your code correctly (i.e., prettier cleaned it up)
- [x] Documented any new components or features
- [x] Added any changes in this PR to the `CHANGELOG.md` `Next` section
- [x] If this pull request includes a new component or feature, has it been exported from one of the library's entry points?

### CODE REVIEW

File: src/lib/SVGMap/ZoomControls.svelte [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  .zoom-button {
+    height: 30px;
+    width: 30px;
+    appearance: none;
+    border: solid 1px var(--color-gray-shade-light);
+    background: var(--color-white);
+    color: var(--color-gray-shade-darkest);
+    display: flex;
+    align-items: center;
+    justify-content: center;
```
Comment (benkates) [inline src/lib/SVGMap/ZoomControls.svelte:115]: consider adding `cursor: pointer;`

File: src/lib/SVGMap/SVGMap.stories.svelte [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+<Story name="Feature highlight">
+  <SVGMap features={cleveland_bike_data.features} projection={geoMercator}>
+    <PolygonLayer
+      fill={(d) => clevelandBikeScale(d.properties.bike_to_work)}
+      stroke="white"
+      hoverStroke={urbanColors.yellow}
+      hoverStrokeWidth={2}
+      highlightFeature={(d) => d.properties.GEOID === clevelandHighlight}
+      on:click={(e) => (clevelandHighlight = e.detail.props.GEOID)}
```
Comment (benkates) [inline src/lib/SVGMap/SVGMap.stories.svelte:135]: What do you think about "clicking on active element toggles it off"?
```
on:click={(e) => {
  if (clevelandHighlight === e.detail.props.GEOID) {
    clevelandHighlight = null;
  } else {
    clevelandHighlight = e.detail.props.GEOID;
  }
}}
```

File: src/lib/SVGMap/SVGMap.svelte [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  });
+
+  function setupZoom(el) {
+    // create d3 selection of the svg
+    svgSelection = select(el.querySelector("svg"));
+    // create zoom instance
+    mapZoom = zoom()
+      .scaleExtent([1, 8])
+      .filter((event) => {
+        if (scrollWheel === "ctrl") {
```
Comment (benkates) [inline src/lib/SVGMap/SVGMap.svelte:84]: consider mac users with `alt` key as well (maybe it's both ctrl or alt if ctrl is selected?) - related: mapbox's [cooperative gestures](https://docs.mapbox.com/mapbox-gl-js/example/cooperative-gestures/)

File: src/lib/SVGMap/PolygonLayer.svelte [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+  function getAriaLabel(feature) {
+    if (typeof ariaLabel === "string" || typeof ariaLabel === "undefined") {
+      return ariaLabel;
+    }
+    return ariaLabel(feature);
+  }
+
+  function handleMousemove(e, feature) {
+    raise(e.target);
```
[the remark this answers is not in the document]
Comment (mitchthorson) [reply src/lib/SVGMap/PolygonLayer.svelte:100]: yeah agree here, this was bugging me, I'll look into raising both

File: src/lib/SVGMap/SVGMap.svelte (hunk 3 above)
Comment (mitchthorson) [reply src/lib/SVGMap/SVGMap.svelte:84]: ah yeah interesting. i feel like just the `ctrl` approach is a wider standard (Datawrapper works with just `ctrl`), but i do kinda like the `command` option for mac users...probably no downside to allowing both

File: src/lib/SVGMap/ZoomControls.svelte [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+   * Where the zoom control UI should be positioned.
+   * @type {"bottom-left" | "bottom-right" | "top-left" | "top-right"}
+   */
+  export let controlPosition;
+
+  $: top = controlPosition.includes("top");
+</script>
+
+<div class="zoom-controls {controlPosition}">
+  {#if showReset && top}
```
Comment (benkates) [inline src/lib/SVGMap/ZoomControls.svelte:16]: Nice to have would be only showing the reset button when the coords/zoom level do not equal those of the "true" reset. Right now it shows even if you zoom in/back out and don't pan.

In a previous mapbox project, I was logging the coords/zoom level on load (when the map was all the way zoomed out). Then I only showed the reset icon if the coords didn't match. Just sharing as an idea here!

File: src/lib/SVGMap/SVGMap.svelte [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+  });
+
+  // set up a store to hold the scaled projection function
+  const projectionStore = writable(projectionFn);
+  // update the store any time the projection function updates
+  $: projectionStore.set(projectionFn);
+
+  // initialize a transform store in case zoom is turned on
+  // this will be where we store the output of the d3-zoom behaviour and broadcast it
+  let transformStore = writable(zoomIdentity);
```
Comment (benkates) [inline src/lib/SVGMap/SVGMap.svelte:62]: Not sure how with this setup, but a nice to have would be setting CSS cursor property to "grab"/"grabbing" before/during drag.

File: src/lib/SVGMap/ZoomControls.svelte [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        <path
+          fill-rule="evenodd"
+          clip-rule="evenodd"
+          d="M0 15L-2.62268e-07 9L2 9L2 13L6 13L6 15L0 15Z"
+          fill="black"
+        />
+      </svg>
+    </button>
+  {/if}
+  <button class="zoom-button zoom-out" on:click={zoomOut}>
```
Comment (benkates) [inline src/lib/SVGMap/ZoomControls.svelte:47]: consider making this disabled on max zoom

File: src/lib/SVGMap/SVGMap.stories.svelte (hunk 2 above)
Comment (mitchthorson) [reply src/lib/SVGMap/SVGMap.stories.svelte]: added!

File: src/lib/SVGMap/PolygonLayer.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (mitchthorson) [reply src/lib/SVGMap/PolygonLayer.svelte:105]: fixed!

File: src/lib/SVGMap/ZoomControls.svelte (hunk 5 above)
Comment (mitchthorson) [reply src/lib/SVGMap/ZoomControls.svelte:16]: that is indeed how this works, the trick is zooming in and out don't always leave you _exactly_ where you started. 

logic here: https://github.com/UrbanInstitute/dataviz-components/pull/89/files#diff-494280830062b5a0c634081adaa111d5dddc505c2c92cc05c7a874316b00b6f7R154

File: src/lib/SVGMap/SVGMap.svelte (hunk 6 above)
Comment (mitchthorson) [reply src/lib/SVGMap/SVGMap.svelte:62]: adding this as an idea to the backlog!

File: src/lib/SVGMap/ZoomControls.svelte (hunk 5 above)
Comment (mitchthorson) [reply src/lib/SVGMap/ZoomControls.svelte:16]: actually i needed to get a little more specific with the logic! working better now

File: src/lib/SVGMap/ZoomControls.svelte (hunk 7 above)
Comment (mitchthorson) [reply src/lib/SVGMap/ZoomControls.svelte]: good call here, added this in for both max and min zoom

File: src/lib/SVGMap/SVGMap.svelte (hunk 3 above)
Comment (mitchthorson) [reply src/lib/SVGMap/SVGMap.svelte:114]: ok, `metaKey` is now supported in addition to `ctrlKey` for this.

**Comment by mitchthorson** [comment]
@benkates thanks for your ideas! I've addressed most of them and put the rest into the backlog for future improvements (mostly just the drag cursors, as those might take a bit more work to implement). I've also reorged a bit and given the maps their own root-level export. Feel free to take another gander at everything here and let me know if anything is amiss.

**Comment by mitchthorson** [comment]
Including these changes in a unified PR: #92

**Pull request closed without merging**